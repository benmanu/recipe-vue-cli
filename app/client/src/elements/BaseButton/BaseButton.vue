<template>
  <button>{{ text }}</button>
</template>

<script>
export default {
  name: 'BaseButton',
  props: {
    text: {
      type: String,
      default: 'Button',
    },
  },
};
</script>

<style scoped lang="less">
</style>
