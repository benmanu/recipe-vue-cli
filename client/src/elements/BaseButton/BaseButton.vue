<template>
  <button>{{ text }}</button>
</template>

<script>
export default {
  name: 'BaseButton',
  props: {
    text: String,
  },
};
</script>

<style scoped lang="less">
</style>
