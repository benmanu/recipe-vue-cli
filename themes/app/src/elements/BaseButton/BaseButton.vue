<template>
  <button @click="onClick">{{ text }}</button>
</template>

<script>
export default {
  name: 'BaseButton',
  props: {
    text: {
      type: String,
      default: 'Button',
    },
    onClick: {
      type: Function,
      default: () => {},
    },
  },
};
</script>

<style scoped lang="less">
</style>
