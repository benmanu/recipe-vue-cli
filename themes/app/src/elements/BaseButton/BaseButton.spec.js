import { shallowMount } from '@vue/test-utils';
import BaseButton from './BaseButton.vue';

describe('BaseButton.vue', () => {
  it('renders props.text when passed', () => {
    const text = 'new button';
    const wrapper = shallowMount(BaseButton, {
      propsData: { text },
    });
    expect(wrapper.text()).toMatch(text);
  });

  it('renders correctly', () => {
    const wrapper = shallowMount(BaseButton);
    expect(wrapper).toMatchSnapshot();
  });
});
