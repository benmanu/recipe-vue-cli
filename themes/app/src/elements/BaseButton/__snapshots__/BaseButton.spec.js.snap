// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton.vue renders correctly 1`] = `<button>Button</button>`;
